newspassidBidAdapter.js

```diff
@@ -0,0 +1,672 @@
+    extObj['newspassid']['np_rw'] = placementIdOverrideFromGetParam ? 1 : 0;
+    if (validBidRequests.length > 0) {
+      let userIds = this.cookieSyncBag.userIdObject; // 2021-01-06 - slight optimisation - we've already found this info
+      if (userIds.hasOwnProperty('pubcid')) {
+        extObj['newspassid'].pubcid = userIds.pubcid;
+      }
+    }
+    extObj['newspassid'].pv = this.getPageId(); // attach the page ID that will be common to all auciton calls for this page if refresh() is called
+    let whitelistAdserverKeys = config.getConfig('newspassid.np_whitelist_adserver_keys');
+    let useWhitelistAdserverKeys = isArray(whitelistAdserverKeys) && whitelistAdserverKeys.length > 0;
+    extObj['newspassid']['np_kvp_rw'] = useWhitelistAdserverKeys ? 1 : 0;
+    if (getParams.hasOwnProperty('npf')) { extObj['newspassid']['npf'] = getParams.npf === 'true' || getParams.npf === '1' ? 1 : 0; }
+    if (getParams.hasOwnProperty('nppf')) { extObj['newspassid']['nppf'] = getParams.nppf === 'true' || getParams.nppf === '1' ? 1 : 0; }
+    if (getParams.hasOwnProperty('nprp') && getParams.nprp.match(/^[0-3]$/)) { extObj['newspassid']['nprp'] = parseInt(getParams.nprp); }
+    if (getParams.hasOwnProperty('npip') && getParams.npip.match(/^\d+$/)) { extObj['newspassid']['npip'] = parseInt(getParams.npip); }
+    if (this.propertyBag.endpointOverride != null) { extObj['newspassid']['origin'] = this.propertyBag.endpointOverride; }
+    var userExtEids = this.generateEids(validBidRequests); // generate the UserIDs in the correct format for UserId module
+    npRequest.site = {
+      'publisher': {'id': htmlParams.publisherId},
+      'page': document.location.href,
+      'id': htmlParams.siteId
+    };
+    npRequest.test = (getParams.hasOwnProperty('pbjs_debug') && getParams['pbjs_debug'] === 'true') ? 1 : 0;
+    if (bidderRequest && bidderRequest.uspConsent) {
+      logInfo('ADDING USP consent info');
+      deepSetValue(npRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    } else {
+      logInfo('WILL NOT ADD USP consent info; no bidderRequest.uspConsent.');
+    }
+    if (schain) { // we set this while iterating over the bids
+      logInfo('schain found');
+      deepSetValue(npRequest, 'source.ext.schain', schain);
+    }
+    if (config.getConfig('coppa') === true) {
+      deepSetValue(npRequest, 'regs.coppa', 1);
+    }
+    if (singleRequest) {
+      logInfo('buildRequests starting to generate response for a single request');
+      npRequest.id = bidderRequest.auctionId; // Unique ID of the bid request, provided by the exchange.
+      npRequest.auctionId = bidderRequest.auctionId; // not sure if this should be here?
+      npRequest.imp = tosendtags;
+      npRequest.ext = extObj;
+      deepSetValue(npRequest, 'source.tid', bidderRequest.auctionId);// RTB 2.5 : tid is Transaction ID that must be common across all participants in this bid request (e.g., potentially multiple exchanges).
```

Comment:
transactionid

Reply:
not sure I follow this comment - what are you referring too ?

Reply:
I'm also unsure what you mean here @patmmccann , It should be .tid, shouldn't it?

Reply:
we discussed in committee, auction id is correct